### PR TITLE
Road to 0.16.0

### PR DESCRIPTION
**Pull request by signorecello**
Closing a bunch of PRs in order to meet with 0.16.0

### CODE REVIEW

File: docs/language_concepts/data_types/03_strings.md [hunk 1]
```diff
@@ -41,3 +41,23 @@ fn main() {
...
+| `\t`            | Tab             |
+| `\0`            | Null Character  |
+| `\"`            | Double Quote    |
+| `\\`            | Backslash       |
+
+Example:
+
+```rust
+let s = "Hello \"world" // prints "Hello "world"
+let s = "hey \tyou"; // prints "Hello   world"
```
Comment (critesjosh) [inline docs/language_concepts/data_types/03_strings.md:62]:
```suggestion
let s = "hey \tyou"; // prints "hey   you"
```

File: docs/language_concepts/data_types/03_strings.md [hunk 2]
```diff
@@ -41,3 +41,23 @@ fn main() {
     assert(message_bytes[0] == message_vec.get(0));
 }
 ```
+
+## Escape characters
+
+You can use escape characters for your strings, as much as escape sequences:
```
Comment (critesjosh) [inline docs/language_concepts/data_types/03_strings.md:47]: I think this phrasing is confusing. Can you just remove "as much as escape sequences"?

File: docs/language_concepts/data_types/06_vectors.mdx [hunk 3]
```diff
@@ -21,3 +21,152 @@ for i in 0..5 {
...
+
+Creates a new, empty vector.
+
+```rust
+pub fn new() -> Self {
+    Self { slice: [] }
+}
+```
+
+example:
```
Comment (critesjosh) [inline docs/language_concepts/data_types/06_vectors.mdx:37]:
```suggestion
Example:
```

File: docs/language_concepts/data_types/06_vectors.mdx [hunk 4]
```diff
@@ -21,3 +21,152 @@ for i in 0..5 {
...
+
+Creates a vector containing each element from a given slice. Mutations to the resulting vector will not affect the original slice.
+
+```rust
+pub fn from_slice(slice: [T]) -> Self {
+    Self { slice }
+}
+```
+
+example:
```
Comment (critesjosh) [inline docs/language_concepts/data_types/06_vectors.mdx:54]:
```suggestion
Example:
```

File: docs/language_concepts/data_types/06_vectors.mdx [hunk 5]
```diff
@@ -21,3 +21,152 @@ for i in 0..5 {
...
+
+Retrieves an element from the vector at a given index. Panics if the index points beyond the vector's end.
+
+```rust
+pub fn get(self, index: Field) -> T {
+    self.slice[index]
+}
+```
+
+example:
```
Comment (critesjosh) [inline docs/language_concepts/data_types/06_vectors.mdx:72]:
```suggestion
Example:
```

File: docs/language_concepts/data_types/06_vectors.mdx [hunk 6]
```diff
@@ -21,3 +21,152 @@ for i in 0..5 {
...
+
+Adds a new element to the vector's end, returning a new vector with a length one greater than the original unmodified vector.
+
+```rust
+pub fn push(&mut self, elem: T) {
+    self.slice = self.slice.push_back(elem);
+}
+```
+
+example:
```
Comment (critesjosh) [inline docs/language_concepts/data_types/06_vectors.mdx:89]:
```suggestion
Example:
```

File: docs/language_concepts/data_types/06_vectors.mdx [hunk 7]
```diff
@@ -21,3 +21,152 @@ for i in 0..5 {
...
+
+```rust
+pub fn pop(&mut self) -> T {
+    let (popped_slice, last_elem) = self.slice.pop_back();
+    self.slice = popped_slice;
+    last_elem
+}
+```
+
+example:
```
Comment (critesjosh) [inline docs/language_concepts/data_types/06_vectors.mdx:109]:
```suggestion
Example:
```

File: docs/language_concepts/data_types/06_vectors.mdx [hunk 8]
```diff
@@ -21,3 +21,152 @@ for i in 0..5 {
...
+
+Inserts an element at a specified index, shifting subsequent elements to the right.
+
+```rust
+pub fn insert(&mut self, index: Field, elem: T) {
+    self.slice = self.slice.insert(index, elem);
+}
+```
+
+example:
```
Comment (critesjosh) [inline docs/language_concepts/data_types/06_vectors.mdx:128]:
```suggestion
Example:
```

File: docs/language_concepts/data_types/06_vectors.mdx [hunk 9]
```diff
@@ -21,3 +21,152 @@ for i in 0..5 {
...
+
+```rust
+pub fn remove(&mut self, index: Field) -> T {
+    let (new_slice, elem) = self.slice.remove(index);
+    self.slice = new_slice;
+    elem
+}
+```
+
+example:
```
Comment (critesjosh) [inline docs/language_concepts/data_types/06_vectors.mdx:148]:
```suggestion
Example:
```

File: docs/language_concepts/data_types/06_vectors.mdx [hunk 10]
```diff
@@ -21,3 +21,152 @@ for i in 0..5 {
...
+
+Returns the number of elements in the vector.
+
+```rust
+pub fn len(self) -> Field {
+    self.slice.len()
+}
+```
+
+example:
```
Comment (critesjosh) [inline docs/language_concepts/data_types/06_vectors.mdx:167]:
```suggestion
Example:
```

File: docs/migration_notes.md [hunk 11]
```diff
@@ -6,6 +6,16 @@ keywords: [Noir, notes, migration, updating, upgrading]
 
 Noir is in full-speed development. Things break fast, wild, and often. This page attempts to leave some notes on errors you might encounter when upgrading and how to resolve them until proper patches are built.
 
+## ≥0.14
+
+The index of the [for loops](./language_concepts/02_control_flow.md#loops) is now of type `u64` instead of `Field`. An example refactor would be:
```
Comment (critesjosh) [inline docs/migration_notes.md:11]: Can we silence the warnings from the compiler about this yet?

File: docs/nargo/02_testing.md [hunk 12]
```diff
@@ -40,3 +40,22 @@ fn test_add() {
     assert(add(2,2) == 5);
 }
 ```
+
+You can be more specific and make it fail with a specific reason by using `should_fail_with = "<the reason for failure>`:
+
+```rust
+fn main(african_swallow_avg_speed : Field) {
+    assert(african_swallow_avg_speed == 65, "What is the airspeed velocity of an unladen swallow");
```
Comment (critesjosh) [inline docs/nargo/02_testing.md:48]: :joy:

**Review by critesjosh — COMMENTED** [review]
lgtm so far, just some minor nits. let me know if you have more stuff to add and would like me to review again.

File: docs/migration_notes.md (hunk 11 above)
Comment (signorecello) [reply docs/migration_notes.md:11]: Just checked, and yes the compiler stops warning about it from 0.14.0 onwards

**Pull request merged**